controller/cache/cache.go

```diff
@@ -161,6 +165,9 @@ type ResourceInfo struct {
 	NodeInfo *NodeInfo
 
 	manifestHash string
+
+	// annotations stores all the ObjectRef annotations
+	annotations map[string]string
```

Comment:
I think we can save memory here and just have a bool instead. annotations like `kubectl.kubernetes.io/last-applied-configuration` can end up consuming a lot if you consider the number of resources argo watches